src/backend/distributed/planner/multi_planner.c

```diff
@@ -75,12 +76,23 @@ CreatePhysicalPlan(Query *parse)
 	Query *parseCopy = copyObject(parse);
 	MultiPlan *physicalPlan = NULL;
 	CmdType commandType = parse->commandType;
+	bool routerPlannable = false;
```

Comment:
I'd be happier if we moved decision for `routerPlannable` flag to a function. In such way, the `CreatePhysicalPlan ` would be more readable.  But, feel free to  overlook this.